src/test/java/mate/academy/jvbookstore/controller/UserControllerTest.java

```diff
@@ -0,0 +1,139 @@
+            @Autowired UserRepository userRepository) {
+        List<User> usersFromDb = userRepository.findAll();
+        List<User> filtered = usersFromDb.stream()
+                .filter(user -> user.getFirstName().equals("owner") 
+                && user.getEmail().equals("owner@bookstore.com"))
+                .toList();
+        Assertions.assertEquals(1, filtered.size());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void findAll_WithPageable_ListOfUsers() throws Exception {
+        Pageable pageable = PageRequest.of(0, 10);
+
+        MvcResult result = mockMvc.perform(get("/users")
+                .content(objectMapper.writeValueAsString(pageable))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        UserResponseDto[] actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), UserResponseDto[].class);
+        
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(2, actual.length);
+        Assertions.assertEquals(actual[0].getEmail(), "owner@bookstore.com");
+        Assertions.assertEquals(actual[1].getEmail(), userFromDb.getEmail());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void updateRoles_WithUserRole_UpdatedUser(
+            @Autowired RoleRepository roleRepository) throws Exception {
+        final Role userRole = roleRepository.findById(1L).get();
+        UserResponseDto expected = new UserResponseDto();
+        expected.setId(userFromDb.getId());
+        expected.setEmail(userFromDb.getEmail());
+        expected.setFirstName(userFromDb.getFirstName());
+        expected.setLastName(userFromDb.getLastName());
+        expected.setShippingAddress(userFromDb.getShippingAddress());
+        List<Role> roles = new ArrayList<>();
+        roles.add(userRole);
+        expected.setRoles(roles);
+
+        MvcResult result = mockMvc.perform(patch("/users/roles/" + userFromDb.getId())
+                .content(objectMapper.writeValueAsString(List.of(userRole)))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        UserResponseDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), UserResponseDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @AfterAll
+    static void clearDb(@Autowired DataSource dataSource) throws SQLException {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("db/user/delete-test-user.sql"));
+        }
+    }
```

Comment:
if you want to test making changes in DB, it's not required to execute SQL scripts for restoring DB after tests, you can use `@Transactional @Rollback` on method annotations or test containers